.gitignore

```diff
@@ -87,3 +84,11 @@ examples/contrib/obj
 examples/dotnet/bin
 examples/dotnet/obj
 
+# cmake code gen (assumes -Bbuild)
+.cmake/
+CMakeCache.txt
+CMakeFiles
+CMakeLists.txt.user
+DartConfiguration.tcl
```

Comment:
are you sure of this DartConfiguration.tcl ? never saw it, out of curiosity which IDE did you use ?

Reply:
this appears to be an artifact generated by cmake to support ctest: https://gitlab.kitware.com/cmake/community/-/wikis/doc/ctest/Generating-Testing-Files 

I am using VisualStudio Code connected to a remote checkout on a Debian system. Pretty sure that VS is not involved but will dig a little deeper.

Reply:
I did a clean clone of or-tools on my Debian system from command line without VisualStudio running. The DartConfiguration.tcl file did not appear after 'cmake -S. -Bbuild  -DBUILD_DEPS:BOOL=ON', but was present in the build subdirectory after 'cmake --build build'. VS is exonerated!

Curiously the -enable_testing() statement which the CTest docs describe for creating the test build targets is removed from CMakeLists.txt by either scip-7.0.1.patch, and/or ZLIB.patch. It looks like this is a new symptom of an ongoing campaign to eliminate CTest!

I have a Dart.cmake template file in the directory where I checked out cmake 3.21 for local build, since Debian comes with a version that is too old to meet the or-tools minimum requirement. My theory is that my locally built cmake is inserting the .tcl file into the build, while your version/configuration of cmake is not.

I submit that including DartConfiguration.tcl in the cmake section of .gitignore can be useful and does no apparent harm.

Reply:
Me again. Wrong about exonerating VS.

Opened VS Code, and now I have a new copy of DartConfiguration.tcl in the root of or-tools, along with the one that was created in .build by the command line cmake.

So the .gitignore entry is **not needed** for the command line one (it is already eliminated by *build*/*), but **is needed** if VisualStudio touches CMake (mine does because of installed extensions).

Which makes an argument to move DartConfiguration.tcl to the IDE section. Although it still is a CMake artifact. @Mizux what do you prefer?

Reply:
Since DartConfiguration.tcl is both a file generated by cmake and a file touch by vscode (CMake extension ?) let's keep it here (aka CMake group take precedence) -> I'll merge this PR as it